driver/convert.c

```diff
@@ -4255,10 +4257,80 @@ static SQLRETURN struct_to_iso8601_timestamp(esodbc_stmt_st *stmt,
 	return SQL_SUCCESS;
 }
 
+/* apply corrections depending on the (column) size and decimal digits
+ * values given at binding time: nullify or trim the resulted string:
+ * https://docs.microsoft.com/en-us/sql/odbc/reference/appendixes/column-size
+ * */
+static SQLRETURN size_decdigits_for_iso8601(esodbc_rec_st *irec,
+	SQLULEN *_colsize, SQLSMALLINT *_decdigits)
+{
+	SQLULEN colsize;
+	SQLSMALLINT decdigits;
+	esodbc_stmt_st *stmt = HDRH(irec->desc)->stmt;
+
+	colsize = get_param_size(irec);
+	DBGH(stmt, "requested column size: %llu.", colsize);
+
+	decdigits = get_param_decdigits(irec);
+	DBGH(stmt, "requested decimal digits: %llu.", decdigits);
+	if (ESODBC_MAX_SEC_PRECISION < decdigits) {
+		WARNH(stmt, "requested decimal digits adjusted from %hd to %d (max).",
+			decdigits, ESODBC_MAX_SEC_PRECISION);
+		decdigits = ESODBC_MAX_SEC_PRECISION;
+	}
+
+	switch (irec->es_type->data_type) {
+		case SQL_TYPE_TIME:
+			if (colsize) {
+				if (colsize < TIME_TEMPLATE_LEN(0) ||
+					colsize == TIME_TEMPLATE_LEN(1) - 1 /* `:ss.`*/) {
+					ERRH(stmt, "invalid column size value: %llu; allowed: "
+						"8 or greater than 9");
+					RET_HDIAGS(stmt, SQL_STATE_HY104);
+				}
+				colsize += DATE_TEMPLATE_LEN + /* ` `/`T` */1;
+			}
+			break;
+		case SQL_TYPE_DATE:
+			/* if origin is a timestamp (struct or string), the time part
+			 * needs to be zeroed. */
+			if (colsize) {
+				if (colsize != DATE_TEMPLATE_LEN) {
+					ERRH(stmt, "invalid column size value: %llu; allowed: %d.",
+						DATE_TEMPLATE_LEN);
+					RET_HDIAGS(stmt, SQL_STATE_HY104);
+				}
+				colsize += /* ` `/`T` */1 + TIME_TEMPLATE_LEN(0);
+			}
+			if (decdigits) {
+				ERRH(stmt, "invalid decimal digits %hd for TIME type.",
+					decdigits);
+				RET_HDIAGS(stmt, SQL_STATE_HY104);
+			}
+			break;
+		case SQL_TYPE_TIMESTAMP:
+			if (colsize && (colsize < TIMESTAMP_NOSEC_TEMPLATE_LEN ||
+					colsize == 17 || colsize == 18)) {
+				ERRH(stmt, "invalid column size value: %llu; allowed: "
+					"16, 19, 20+f.", colsize);
```

Comment:
Is the `f` a typo here?  Also, the message for time was `8 or greater than 9`, so for consistency could this one be `16 or greater than 18`?

Reply:
I've rephrased that. `+f` (lifted from the spec) is now ` + fractions count`.
The size can't be 18 either (it stands for timestamps of formats: `yyyy-MM-dd hh:mm`, `yyyy-MM-dd hh:mm:ss` or `yyyy-MM-dd hh:mm:ss.{f}+`)